### DISCUSSION TITLE
Selecting multiple lipids with different head groups

### DISCUSSION THREAD

**Discussion by Saleh-OM4R** [p-j-smith/lipyphilic#81] [Q&A] [answered]
Hi,

I've just started using LipPyphilic, and I've been trying to select multiple lipids but with different head groups using this code :

```
# Find which leaflet each lipid is in at each frame
leaflets = AssignLeaflets(
    universe=u,
    lipid_sel= 'resname POPE POPC PSM BMGP and name P C11 C12','resname CHL1 and name O3 C3' # Select headgroup beads in the MARTINI forcefield
)
# Select which frames to use and perform the analysis
leaflets.run(start=None, stop=None, step=None)  # this will use every frame in the trajectory

```

but it doesn't work and I get this error:

> 
>   File "/tmp/ipykernel_22911/1408747556.py", line 4
>     lipid_sel= 'resname POPE POPC PSM BMGP and name P C11 C12','resname CHL1 and name O3 C3' # Select headgroup beads in the MARTINI forcefield
>                                                               ^
> SyntaxError: positional argument follows keyword argument

But, if I use the same selection with u.select_atoms it works fine:

`u.select_atoms('resname POPE POPC PSM BMGP and name P C11 C12','resname CHL1 and name O3 C3')`

So my question is how does the (lipid_sel=) read the lipid selection? I thought it uses the same selection language as MDAnalysis?or is it different?

Cheers,

**Comment by p-j-smith** [comment, marked as answer]
Hi @Saleh-OM4R thanks for trying out lipyphilic! 

Lipyphilic does indeed use the same selection language as MDAnalysis. The issue is that you have two selection strings that you are passing to `lipid_sel`.

In MDAnalysis:
```py
u.select_atoms(
    'resname POPE POPC PSM BMGP and name P C11 C12',
    'resname CHL1 and name O3 C3',
)
```
will create a single atom group that contains all atoms selected by both strings.

If this is indeed what you want, then you need to pass a single string to `AssignLeaflets`:

```py
    lipid_sel ='(resname POPE POPC PSM BMGP and name P C11 C12) or (resname CHL1 and name O3 C3)'
```

If you try to pass two separate strings, then Python will assume that you are trying to pass two arguments - one to `lipid_sel` and another to a different argument. This is why it raises the error `positional argument follows keyword argument`.

If on the other hand you want to pass  'resname POPE POPC PSM BMGP and name P C11 C12' to `lipid_sel` and `resname CHL1 and name O3 C3` to `midplane_sel`, then you need to explicitly pass the second string to `midplane_sel`:

```py
leaflets = AssignLeaflets(
    universe=u,
    lipid_sel= 'resname POPE POPC PSM BMGP and name P C11 C12',
    midplane_sel='resname CHL1 and name O3 C3',
)
```
